### DISCUSSION TITLE
SSL TCP listener

### DISCUSSION THREAD

**Discussion by Wet-Bandit** [nextgenhealthcare/connect#4883] [Q&A]
The customer is requiring that we add SSL to our TCP listener Port with MLLP channels.  Is there a simple way to do this?  And they are not willing to purchase plug-ins or mirth connect licensing.  The burden to make things work is mine.  
We also can't use a 3rd party like Stunnel, it needs to be done at the application level, which means in mirth connect.
Any help is appreciated.

Mirth Connect Server 3.11.0
Built on April 6, 2021
Java version: 1.8.0_181

**Comment by pacmano1** [comment]
ssltunnel or haproxy are 2 common options.

**Reply by pacmano1** [reply]
If you can't do those, are you also saying your employer doesn't have the firewall/switch/loadbalncer technology to implement that for you?   That's absurd, as is their unwillingness to spend a couple of grand a year for either Zen's or NG's SSL extension.   Best wishes with your plugin.

**Reply by Wet-Bandit** [reply]
pacmano1 :  Please delete this message or remove the reference to the customer.  It is against policy to mention them in online forums.  I got into trouble for you mentioning them.

**Reply by pacmano1** [reply]
You are funny.

I am not subject to your employer's policies.    

![snek](https://user-images.githubusercontent.com/44065187/149818219-7eb56219-b327-4b3a-9d4a-29dcfa23abe9.png)

**Reply by Wet-Bandit** [reply]
I asked you to please remove it.  I will be contacting Next Gen mirth connect Corp with the request, if you refuse.

**Reply by pacmano1** [reply]
Actually what you did was this:

On your other thread on this topic you assumed people were talking to you rudely which just wasn't the case.   That is a problem with the written word of course given there is no intonation to facilitate meaning.  I repled with the fact fhat folks weren't really doing that.  AND in fact I proposed yet another solution (IPSec).   I then proceeded to say(not speaking for others on the thread) that I didn't like your tone and was excusing myself from helping.   

Your reply to that was below _a month later,_ yet again confusing rudeness with a difference of opintion.  

> Please only post if you have a working suggestion to the posted problem and scenario. Rude comments and then being defensive because you were called out on being rude is not allowed in the forum.

On this thread I was venting at your employer for effectively what I perceive as their unwillingness to help you do your job.   There is nothing wrong with that and most importantly - _you neither shared your employer on this thread (but do note finding your employer is triviial) nor made the comment_.    

So let me vent again on your employer - they should not call you out on this or any other thread in GitHub since you didn't say who you worked for (until you chose to confirm it here in the discussion which maybe wasn't the best thing for you to do).   

What perhaps makes me most dissappointed in this thread is the fact I was advocating for your employer to help more in the thing it tasks its employees, that's a good thing.   Please do reach out to Nextgen if you need to.    

Lastly, as you now you pulbically post where you work on assorted forums.

**Reply by Wet-Bandit** [reply]
The work policy is not that the entity can't be mentioned, it is that it can't be defamed in a derogatory context or manner by a contractor or an employee.   In the context of criticism where the entity is not in a position to defend itself etc.  Associates are encouraged to address issues blatantly and direct but not to disclose involved entity names directly in the same context or Thread.  I get it, because no business wants to pay a contractor or an employee that is doing public criticism of the business.  We do have rights in this country but we do not have the right to a job.  Meaning, you can get let go for any reason or no reason in many states.  It is not a protected class or behavior by law, hence the reason they make the policy in the first place.  I wish to keep my customer, so please remove the reference.

**Reply by pacmano1** [reply]
Your response doesn't really apply to me.

1) If you criticized your employer, that is on you.   I suggest you revise your posts that did so.   
2) I found out who you worked for by public available information by _your_ participating on public forums, again, that's on you, not on me.   
3) I don't know if your employer is in the position to defend itself - but I don't think there is anything that prevents their feedback on this topic.

Through all this, you have never taken the time to just say "Maybe I was wrong here and misinterpreted what I thought was a rude response".    As you know the folks that help here are not paid to do so.   

Frankly if you simply would say  - My bad, I misunderstood the intention of the participants -  I'd revise my posts to generically refer to "employers" and likely delete the vast majority of my part of this thread/topic.

I do stand by that overall point in all this and I am speaking as a business owner.   I find it bad practice by an employer and perhaps your direct supervisor to make their employees lives difficult by their unwillingness to make a very small purchase or advocate for you for that purchase in the grand scheme of certain budgets.  Sure there are use cases that might make that cost prohibitive usually due to running hundreds of engines at client sites, that is yet another topic.

**Reply by Wet-Bandit** [reply]
RE: No step on snek.  To be honest I didn't even know what that image means or references, but I had a co-worker who said you are referring to freedom of speech rights in America etc.
My reply to that is: The freedom of United States is a delusion.  Access to fiat currency (aka debt notes) is the only real freedom and almost everyone is a debt slave to get a paycheck and hence has no true freedom.  Except of course the freedom to move from one slave owner to another and not be discriminated against if you are a protected class.  Even if you own your own business, you may have a bit more freedom, but you are still a slave to needed income.  Being a slave means you are obligated to follow the rules of the slave owner, not the law of the country.  Unless of course you don't need the paycheck.  And that is the catch.  Did the tobacco company employee (whistle blower) really have freedom of speech when he told people the truth about it's deadly contents?  Does the Milk industry employee even today admit that cow's milk contains the same antigen that's in B+ blood type and is harmful to most humans?  Acne / moles / stomach & bowl problems / intolerance.  If you are not a B+ blood type go drink or eat some concentrated whey protein and see what happens to prove this to yourself.  Did Microsoft replace my kids XBox 360 when it got the red ring of death? Corporate dominance and Oligarchy is not freedom.

**Reply by Wet-Bandit** [reply]
My response refers to anyone who receives a paycheck from an entity other than themselves.  All places have a similar policy.

**Comment by Wet-Bandit** [comment]
Problem resolved using HTTP, see this link...  https://github.com/nextgenhealthcare/connect/discussions/4885

**Reply by rjkroll** [reply]
Man, this thread had a lot of ups-and-downs for a application support thread!   Looks like it was resolved, but to anyone else reading, you can always create a custom java SSL connector using Java, your own cert truststore, and IIS.
